lib/ddtrace/contrib/cucumber/formatter.rb

```diff
@@ -0,0 +1,104 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/ci'
+require 'ddtrace/ext/test'
+require 'ddtrace/contrib/analytics'
+require 'ddtrace/contrib/cucumber/ext'
+
+module Datadog
+  module Contrib
+    module Cucumber
+      # Defines collection of instrumented Cucumber events
+      class Formatter
+        attr_reader :config
+        private :config
+
+        attr_reader :current_feature_span, :current_step_span
+        private :current_feature_span, :current_step_span
+
+        def initialize(config)
+          @config = config
+
+          bind_events(config)
+        end
+
+        def bind_events(config)
+          config.on_event :test_case_started, &method(:on_test_case_started)
+          config.on_event :test_case_finished, &method(:on_test_case_finished)
+          config.on_event :test_step_started, &method(:on_test_step_started)
+          config.on_event :test_step_finished, &method(:on_test_step_finished)
+        end
+
+        def on_test_case_started(event)
+          trace_options = {
+            app: Ext::APP,
+            resource: event.test_case.name,
+            service: configuration[:service_name],
+            span_type: Datadog::Ext::AppTypes::TEST,
+            tags: tags.merge(Datadog.configuration.tags)
+          }
+          @current_feature_span = tracer.trace(Datadog::Ext::AppTypes::TEST, trace_options)
+          @current_feature_span.set_tag(Datadog::Ext::Test::FRAMEWORK, Ext::FRAMEWORK)
+          @current_feature_span.set_tag(Datadog::Ext::Test::NAME, event.test_case.name)
+          @current_feature_span.set_tag(Datadog::Ext::Test::SUITE, event.test_case.location.file)
+          @current_feature_span.set_tag(Datadog::Ext::Test::TYPE, Ext::TEST_TYPE)
+          @current_feature_span.set_tag(Datadog::Ext::Test::SPAN_KIND, Datadog::Ext::AppTypes::TEST)
```

Comment:
Is `span.kind` a special tag to be used by our backend, or just another piece of informational data for the user?

I ask because we normally don't prefix tags with `span.`. Also, `Datadog::Ext::AppTypes::TEST` is already captured under `trace_options` (`span_type: Datadog::Ext::AppTypes::TEST`), so this information is already shipped to the backend.

Reply:
IIRC it's an optional tag, i think it's fine to set it but span.type has more semantic meaning in our backend